Source/RunActivity/Viewer3D/Sky.cs

```diff
@@ -137,12 +137,10 @@ public void PrepareFrame(RenderFrame frame, ElapsedTime elapsedTime)
             // Current solar and lunar position are calculated by interpolation in the lookup arrays.
             // The arrays have intervals of 1200 secs or 20 mins.
             // Using the Lerp() function, so need to calculate the in-between differential
-            float diff = GetCelestialDiff();
             // The rest of this increments/decrements the array indices and checks for overshoot/undershoot.
-            while (Viewer.Simulator.ClockTime >= (oldClockTime + 1200)) // Plus key, or normal forward in time; <CSComment> better so in case of fast forward
+            while (Viewer.Simulator.ClockTime >= (oldClockTime - DaylightOffsetS + 1200)) // Plus key, or normal forward in time; <CSComment> better so in case of fast forward
```

Comment:
The Sky.cs has a brother-file: MSTSSky.cs. Everywhere the Sky.cs is changed, the MSTSSky.cs should follow. For example the pair of these lines are: https://github.com/openrails/openrails/blob/71c4e89ab1cc0bb5110274617cd1fdd16e3dec9c/Source/RunActivity/Viewer3D/MSTSSky.cs#L216-L218

Actually the two files should be united or inreited one from the other, to have the common code pieces in a common class. I think code copying is a poor programming technique, I don't know why these were implemented this way.

Reply:
Thanks, Peter.

Neither do I, but I'll try to re-unite the two files.